tgg/misc_analyses/gnomad_small_variant_list_query.py

```diff
@@ -0,0 +1,2144 @@
+                "gs://hail-common/references/human_g1k_v37.fasta.fai",
+            )
+    else:
+        target = hl.get_reference("GRCh38")
+        chain = "gs://hail-common/references/grch37_to_grch38.over.chain.gz"
+        if not target.has_sequence():
+            target.add_sequence(
+                "gs://hail-common/references/Homo_sapiens_assembly38.fasta.gz",
+                "gs://hail-common/references/Homo_sapiens_assembly38.fasta.fai",
+            )
+    if not source.has_liftover(target):
+        source.add_liftover(chain, target)
+
+    return [source, target]
+
+
+def get_variants_samples(version: str, var_ht: hl.Table) -> hl.Table:
+    """
+    Get Table of samples carrying variants in a variant Table for specified gnomAD version.
+
+    :param version: gnomAD version.
+    :param var_ht: Table of variants to find carriers of.
+    :return: Table of variant carriers.
+    """
+    # Convert variants to interval for pushdown.
+    logger.info("Retrieving samples that have the variant...")
+    intervals = var_ht.aggregate(
+        hl.agg.collect(
+            hl.interval(start=var_ht.locus, end=var_ht.locus, includes_end=True)
+        )
+    )
+    mt = get_gnomad_raw_data(version, intervals)
+
+    # Now filter gnomAD to keep only the correct alleles of interest after splitting.
+    mt = mt.semi_join_rows(var_ht.key_by("locus", "alleles"))
+
+    # Create a table of variants -> sample(s).
+    # This Table contains one entry per variant and sample.
+    # Remove all row/columns annotations -- we don't need them in this table.
+    mt = mt.select_rows().select_cols().select_entries(*ENTRY_FIELDS_TO_KEEP)
+
+    # Only keep non-reference genotypes.
+    return mt.filter_entries(mt.GT.is_non_ref()).entries()
+
+
+def get_samples_ht_from_gnomad_mt(
+    gnomad_mt: hl.MatrixTable,
+    var_ht: hl.Table
+) -> hl.Table:
+    """
+    Get a Table of sample non ref genotypes per variant from a gnomAD MT.
+
+    :param gnomad_mt: gnomAD MT.
+    :param var_ht: Variant HT.
+    :return: Table of sample non ref genotypes per variant.
+    """
+    gnomad_mt = gnomad_mt.select_cols()
+
+    variant_mt = gnomad_mt.semi_join_rows(var_ht.key_by("locus", "alleles"))
+    sample_mt = variant_mt.filter_entries(variant_mt.GT.is_non_ref())
+    sample_ht = sample_mt.annotate_rows(
+        samples=hl.agg.collect(hl.struct(**sample_mt.col, **sample_mt.entry))
+    ).rows()
+
+    return sample_ht
+
+
+def get_regions_ht_for_one_variant(
+    var_ht: hl.Table,
+    variant_samples_ht: hl.Table,
+    gnomad_mt: hl.MatrixTable,
+    row: hl.struct,
+    gnomad_version: str,
+):
+    """
+    Get the regions HT for one variant.
+
+    :param var_ht: Variant HT.
+    :param variant_samples_ht: Variant samples HT.
+    :param gnomad_mt: gnomAD MT.
+    :param row: Row to get regions HT for.
+    :param gnomad_version: gnomAD version.
+    :return: Regions HT for one variant.
+    """
+    # Prefix all variants and genotype fields with 'v2'.
```

Comment:
Maybe note v2 != gnomAD v2 but stands for variant 2....this really confused me the first time I started working on this script. I'd propose a `v_a` and `v_b` naming but know v1,v2 is what Sanna expects and how she processes the data